app/models/pod.rb

```diff
@@ -25,7 +25,9 @@ class Pod < ActiveRecord::Base
 
   class << self
     def find_or_create_by(opts) # Rename this method to not override an AR method
-      u = URI.parse(opts.fetch(:url))
+      url = opts.fetch(:url)
```

Comment:
I would be fine with raising `ArgumentError` when `nil` is passed explicitly, but not silently ignoring it. But then the validations should catch it. In any case, either the specs should stub out behaviour that differs in the test environment to match the behaviour in the production environment, or if the value might end up `nil` in the production environment too, the call should be fixed to not happen in that case.